titus-common-runtime/src/main/java/com/netflix/titus/runtime/connector/kubernetes/KubeApiException.java

```diff
@@ -36,7 +36,7 @@ public KubeApiException(String message, Throwable cause) {
     }
 
     public KubeApiException(ApiException cause) {
-        this(cause.getMessage(), cause);
+        this(String.format("%s: httpStatus=%s, body=%s", cause.getMessage(), cause.getCode(), cause.getResponseBody()), cause);
```

Comment:
Nice! 🥳 Does the Kube API always return `metav1.Status` in this context? It'd be especially slick if we could parse the `Message` field from [this struct](https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/types.go#L650-L680). 😎

Reply:
We are talking http here I  think, so I don't think there is a guarantee we have that (could be just plain text that says "503" or whatever?)

We *could* do a try/except, and if we couldn't read the status, then we just print the full json.
But I think printing the full json is probably ok in this case

Reply:
Yep, good point. LGTM :+1: